src/ir/from_ast/astconv.rs

```diff
@@ -254,7 +249,7 @@ impl<'ctx, 'prog> BuildCtx<'ctx, 'prog> {
             ast::TimeSub::Sym { l, r } => {
                 let l = self.time(l);
                 let r = self.time(r);
-                l.sub(r, self.comp)
+                l.sub(r, &mut self.comp)
```

Comment:
Seems like all uses of comp now need a mutable reference to it? We could consider storing a mutable reference to it in the build context or something

Reply:
added helper function instead.